Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -642,6 +642,32 @@ public struct ObjectId: BSONValue, Equatable, CustomStringConvertible, Codable {
 
 }
 
+/// Extension to allow a UUID to be initialized from a Binary BSONValue.
+extension UUID {
+    // TODO: fill the rest of this out for full BSONValue conformance (SWIFT-295).
+
+    internal init(fromBinary binary: Binary) throws {
+        guard binary.subtype != Binary.Subtype.uuidDeprecated.rawValue else {
+            throw MongoError.bsonDecodeError(message: "Binary subtype \(binary.subtype) is deprecated, " +
+                    "use \(Binary.Subtype.uuid) instead.")
+        }
+        guard binary.subtype == Binary.Subtype.uuid.rawValue else {
+            throw MongoError.bsonDecodeError(message: "Expected a UUID binary type " +
+                    "(\(Binary.Subtype.uuid)), got \(binary.subtype) instead.")
+        }
+
+        let data = binary.data
+        let uuid: uuid_t = (
+                data[0], data[1], data[2], data[3],
+                data[4], data[5], data[6], data[7],
+                data[8], data[9], data[10], data[11],
+                data[12], data[13], data[14], data[15]
+        )
```

Comment:
I didn't find much documentation about `uuid_t` in the [UUID docs](https://developer.apple.com/documentation/foundation/uuid), but I just want to confirm that this logic is agnostic about platform endianness. Do I assume correctly that `uuid_t` is simply an array of 16 characters?

Reply:
Yep. From `NSUUID` [docs](https://developer.apple.com/documentation/foundation/nsuuid):
>Because a UUID is expressed simply as an array of bytes, there are no endianness considerations for different platforms.